Amazon.IonDotnet/Internals/Binary/RawBinaryWriter.cs

```diff
@@ -689,9 +689,10 @@ private void WriteDecimalNumber(decimal value, bool writeTid)
         public void WriteTimestamp(Timestamp value)
         {
             const byte varintNegZero = 0xC0;
-            const short minutePrecision = 3;
-            const short secondPrecision = 4;
-            const short fracPrecision = 5;
+
+            value = new Timestamp(value.DateTimeValue.Year, value.DateTimeValue.Month, value.DateTimeValue.Day,
```

Comment:
Why do we need to instantiate a new Timestamp each time we write one?

Reply:
In the specification of timestamp:

>  The component values in the binary encoding are always in UTC, while components in the text encoding are in the local time! This means that transcoding requires a conversion between UTC and local time.

To implement this, here I apply the offset to value of current timestamp. But since properties of Timestamp.cs (namely DateTimeValue) are readonly, hence reassigning the value with adjusted offset.

Reply:
I'm concerned about the performance impact of allocating a new Timestamp every time we write one. This creates a lot of garbage.

Can we apply the conversion to the fields as they're needed?

Reply:
How about something like:
`var dateTimeValue = value.DateTimeValue`
`                 .AddMinutes(-value.LocalOffset)`
`                 .AddSeconds(Decimal.ToDouble(value.FractionalSecond));`

Only for when `Offset.kind == DateTimeKind.local`

And then using `dateTimeValue.Year / dateTimeValue.day / dateTimeValue.hour ... `